contrib/confluentinc/confluent-kafka-go/kafka/kafka_test.go

```diff
@@ -108,75 +110,101 @@ to run the integration test locally:
         confluentinc/cp-kafka:5.0.0
 */
 
-func TestConsumerPoll(t *testing.T) {
-	if _, ok := os.LookupEnv("INTEGRATION"); !ok {
-		t.Skip("to enable integration test, set the INTEGRATION environment variable")
-	}
-
-	mt := mocktracer.Start()
-	defer mt.Stop()
-
-	// first write a message to the topic
-
-	p, err := NewProducer(&kafka.ConfigMap{
-		"group.id":            testGroupID,
-		"bootstrap.servers":   "127.0.0.1:9092",
-		"go.delivery.reports": true,
-	}, WithAnalyticsRate(0.1))
-	assert.NoError(t, err)
-	delivery := make(chan kafka.Event, 1)
-	err = p.Produce(&kafka.Message{
-		TopicPartition: kafka.TopicPartition{
-			Topic:     &testTopic,
-			Partition: 0,
+func TestConsumerFunctional(t *testing.T) {
+	tests := []struct {
```

Comment:
Since this variable isn't reused anywhere, can you please inline it into the `for` range? It's a nit, but we've been doing the same everywhere.

Reply:
Din't catch your point, sorry.
What variable?

Reply:
Instead of defining `tests` here and then doing `for _, tt := range tests {`, I'd prefer doing `for _, tt := range []struct{`. Just a nit, no biggie.

Reply:
oh, got it.